datadog_lambda/tracing.py

```diff
@@ -241,6 +261,9 @@ def extract_context_from_sqs_or_sns_event_or_context(event, lambda_context):
         msg_attributes = first_record.get("messageAttributes")
         if msg_attributes is None:
             sns_record = first_record.get("Sns") or {}
+            # SNS->SQS event would extract SNS arn without this check
```

Comment:
Just want to double check that this is what you want.

If we have something like Lambda -> SNS -> SQS -> Lambda, do we set the checkpoint using the SNS arn or the SQS arn?

I would think we would set it with the SNS arn since that's the arn we would be setting it with on the produce end. But I don't know DSM well enough to know. @piochelepiotr do you know?

Reply:
we want to use the SQS arn in this case. Otherwise, we want be able to know that data went through sqs.

Data streams will handle that correctly in the backend by using the context that was propagated (so it knows who the parent is)

Reply:
Gotcha, thanks for the clarification!!